File: core/contracts/src/health.ts
/**
 * Health reporting types consumed by the kernel Supervisor.
 */

export type HealthState = 'healthy' | 'degraded' | 'unhealthy';

export interface HealthStatus {
  state: HealthState;
  /** Human-readable detail for the shell / logs. */
  detail?: string;
  /** Epoch millis when this status was produced. */
  checkedAt: number;
}

export function healthy(detail?: string): HealthStatus {
  return { state: 'healthy', detail, checkedAt: Date.now() };
}

export function degraded(detail: string): HealthStatus {
  return { state: 'degraded', detail, checkedAt: Date.now() };
}

export function unhealthy(detail: string): HealthStatus {
  return { state: 'unhealthy', detail, checkedAt: Date.now() };
}

File: core/contracts/python/aah_contracts/health.py
"""Health reporting types (Python mirror of health.ts)."""

from __future__ import annotations

import time
from dataclasses import dataclass
from typing import Literal

HealthState = Literal["healthy", "degraded", "unhealthy"]


@dataclass(frozen=True, slots=True)
class HealthStatus:
    state: HealthState
    checked_at: int
    detail: str | None = None


def _now_ms() -> int:
    return int(time.time() * 1000)


def healthy(detail: str | None = None) -> HealthStatus:
    return HealthStatus(state="healthy", detail=detail, checked_at=_now_ms())


def degraded(detail: str) -> HealthStatus:
    return HealthStatus(state="degraded", detail=detail, checked_at=_now_ms())


def unhealthy(detail: str) -> HealthStatus:
    return HealthStatus(state="unhealthy", detail=detail, checked_at=_now_ms())
